llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -399,6 +399,54 @@ adderGenerator(const int32_t From, const int32_t To, const int32_t StepSize) {
   };
 }
 
+Register CombinerHelper::createUnmergeValue(
```

Comment:
Curious: Maybe `IRTranslator` already has a similar piece of code that we can re-use?

Reply:
Not from what I can see, `G_UNMERGE_VALUES` isn't meant for vectors to begin with. It only supports scalars, according to the documentation. Most backends just support this as an extension since there is no other opcode to do it. In theory, this will be ripped out in favour of a simple `G_SUBREGISTER_EXTRACT` in the near future if that opcode gets proper support. `Utils.cpp` does have something that is a bit similar, called `llvm::extractParts`, but that splits into two parts while this code goes through it like a binary tree.

In any case, I would keep it like this since this code should be replaced before upstreaming and that is easier if we are the only ones depending on it.